visualizations/plotly/tests/test_filtered_plotly.py

```diff
@@ -61,6 +62,20 @@ def testNonStringLabels(self):
                         for label in labels)
                         for labels in itervalues(filters)))
 
+    def testSendDataToCloudRemoved(self):
+        filtered = MockElement(self.data)
+
+        self.assertTrue('modeBarButtonsToRemove' in filtered['config'])
+        self.assertTrue('sendDataToCloud' in
+                        filtered['config']['modeBarButtonsToRemove'])
+
+    def testModeBarButtonsRemovalWarning(self):
+        with warnings.catch_warnings(record=True) as w:
+            MockElement(self.data, config={'modeBarButtonsToRemove': []})
+
+            assert len(w) == 2
+            assert 'required' in str(w[-1].message)
```

Comment:
This should use self.assertTrue. Could be a good idea to also check that the warning includes the required removed buttons:

```
self.assertTrue(any(button in w.message for Plotly.dissallowed_buttons))
```